src/conversion/mapper.js

```diff
@@ -104,33 +172,35 @@ export default class Mapper {
 	/**
 	 * Gets the corresponding model position.
 	 *
+	 * @fires engine.conversion.Mapper#event:viewToModelPosition
 	 * @param {engine.view.Position} viewPosition View position.
 	 * @returns {engine.model.Position} Corresponding model position.
 	 */
 	toModelPosition( viewPosition ) {
-		let viewBlock = viewPosition.parent;
-		let modelParent =  this._viewToModelMapping.get( viewBlock );
-
-		while ( !modelParent ) {
-			viewBlock = viewBlock.parent;
-			modelParent = this._viewToModelMapping.get( viewBlock );
-		}
+		const data = {
+			modelPosition: null
+		};
 
-		let modelOffset = this._toModelOffset( viewPosition.parent, viewPosition.offset, viewBlock );
+		this.fire( 'viewToModelPosition', viewPosition, data );
```

Comment:
At the beginning, I thought about it and did it like you said. But then I thought that it is clearer this way. Still, if you really want you can change parent/offset of `viewPosition` but it is much less explicit that this variable can be changed. If you really want I will move `viewPosition` to `data`.